scripts/generate.js

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env node
+const fs = require('fs/promises');
+
+const buildPluginContent = (pluginName) =>
+  `import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType } from '../../common/types';
+
+export class ${pluginName} extends Plugin {
+  public name: string = '${pluginName}';
+  public description: string = 'Some sort of a description.';
+  public usage: string = '${pluginName}';
+  public permission: ChannelType = ChannelType.Public;
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public async execute(message: IMessage, args?: string[]) {
+    // Your logic here.
+  }
+}
+`;
+
+const buildPluginFriendlyName = (name) =>
+  name
+    .split(' ')
+    .filter(Boolean)
+    .map((token) => `${token.substr(0, 1).toUpperCase()}${token.substr(1)}`)
+    .join('')
+    .concat('Plugin');
+
+const buildPluginLoader = (pluginLoader, pluginName, fileName) => {
+  const tokens = pluginLoader.split('\n');
+  const desiredIdx = tokens.findIndex((r) => r.length === 0);
+  tokens.splice(
+    desiredIdx,
+    0,
+    `import { ${pluginName} } from '../app/plugins/${fileName}.plugin';`
+  );
+  return tokens.join('\n').replace(
+    `};`,
+
+    `  ${pluginName.toLowerCase().substr(0, pluginName.length - 6)}: ${pluginName},
+};`
+  );
+};
```

Comment:
bit sketchy

maybe we should put a comment in `plugin.loader.ts` that says 'do not edit; file managed by generate.js'

Reply:
we can create a parser to confirm that `plugin.loader.ts` is in the correct format, have it test on circleci, and break build if it isn't.

Reply:
agree